core/src/test/java/forklift/consumer/injection/ConsumerTest.java

```diff
@@ -127,6 +134,63 @@ public void injectJson() {
         assertEquals("default", ec.strval);
     }
 
+    /**
+     * Tests constructor injection into a consumer.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void jsonConstructorInjection() throws Exception {
+        ConsumerService service = new ConsumerService(ServiceBeanResolver.class);
+        Consumer test = new Consumer(ConstructorJsonConsumer.class, forklift, this.getClass().getClassLoader());
+        test.setServices(Arrays.asList(service));
```

Comment:
I think it's worth considering a refactor where the injection code is removed from the consumer entirely(this is what I started to do initially).  

The setServices method was already in place and I tried to stick with adding the new functionality and not go down a refactoring rabbit hole.